docs/usage.md

```diff
@@ -249,3 +266,38 @@ initializes and makes use of TLS. If you're seeing compartment violations when
 accessing TLS then you likely need to run `pad-tls` on the relevant. The tool
 can be found at `$IA2_PATH/build/tools/pad-tls/pad-tls`. Note that the `ldd` and
 `lddtree` tools can be used to list the DSO dependencies of your app.
+
+## Hidden Symbols and Linker Errors
+
+When building the post-rewriter compartmentalized version of your project you
+may run into linker errors where `libcallgates.so` can't find symbols within a
+compartment. There are a couple of reasons why that may happen.
+
+As mentioned in the rewriter section, if a function is `static`, but the
+`static` keyword comes from inside a macro, then the rewriter will fail to emit
+an `__attribute__((used))` annotation for the function. This can lead to linker
+errors if the function is used as an address-taken function across compartments.
+
+Similar issues can occur when using linker scripts. If your project uses a
+linker script to mark functions local, you'll run into linker errors where the
+generated callgates can't find the local function. For example if you had a
+linker script:
+
+```txt
+YOURLIB_1.0.0 {
+  local:
+    foo;
+};
+```
+
+If you're seeing errors along the lines of:
+
+```txt
+/usr/bin/ld: libcallgates.so: undefined reference to `foo'
+```
+
+Then IA2 needs `foo` to not be marked local.
+
+Note also that an `__attribute__((visibility ("hidden")))` annotation can have
+the same effect as hiding the symbol in a linker script. Such annotation will
+have to be removed to work in a containerized app.
```

Comment:
We figured out how to handle hidden symbols in #451.  Though it appears #435 was then re-opened, but I forget why (@ayrtonm, do you remember?).

Reply:
It was closed by mistake. There's some edge case we still don't support but I don't remember at the moment. IIRC adding the flag to the test suite should point it out.

Reply:
Hrm, interesting I didn't realize we had already resolved part of this. I'm trying to reproduce the issues I'm documenting here and it seems like maybe these docs are now redundant?

* `__attribute__((used))` is now being inserted correctly for static functions when `static` comes from inside a macro.
* Functions marked with `__attribute__((visibility ("hidden")))` now link correctly.

I still need to test if the issue with a version script is also fixed, but if it is I think we can close this PR.

Reply:
Looks like the fix for `visibility("hidden")` also addressed the issue with having a linker script mark a symbol `local`? Or at least the issue was fixed at some point because I can now include the zlib linker script in the compartmentalized build and I no longer hit the linker errors I originally saw. So I'm going to close this and the related bugs.